pkg/ip/link_linux.go

```diff
@@ -60,11 +60,15 @@ func peerExists(name string) bool {
 	return true
 }
 
-func makeVeth(name string, mtu int) (peerName string, veth netlink.Link, err error) {
+func makeVeth(name, vethPeerName string, mtu int) (peerName string, veth netlink.Link, err error) {
```

Comment:
I don't think veth-peer name in host-side can be configurable without some checks, because interface name actually has some constraints, like **length** and **uniqueness**. It's dangerous to put this as an input parameter. I think random name is a safe and general option for us.

Reply:
Exactly as you described that - this randomness should be an option, but having a possibility to set a particular value, as its provided by kernel - is also an option and having this on the lib side (e.g. to enforce some naming scheme) seems to be very valuable.
On the same time I think that validation of these constraints should not be done on lib level. These rules will be enforced by kernel, so if user will provide incorrect values - he will simply receive an error.

btw. what exactly do you see dangerous in leaving such option for an user?

Reply:
Thank you for reviewing, like Piotr said, I think we can rely on kernel. Actually, the uniqueness of current logic is enforced by kernel.